maestro-studio/web/src/api.ts

```diff
@@ -32,8 +32,12 @@ const useRepl = (): ReplResponse => {
 }
 
 export const API = {
-  getDeviceScreen: async (): Promise<DeviceScreen> => {
-    return makeRequest('GET', '/api/device-screen')
+  useDeviceScreen: (config?: SWRConfiguration<DeviceScreen>): SWRResponse<DeviceScreen> => {
+    return useSWR('/api/device-screen?deletePrevious=true', (url) => makeRequest('GET', url), config)
+  },
+  getDeviceScreen: async (deletePrevious?: boolean): Promise<DeviceScreen> => {
+    const query = deletePrevious ? '?deletePrevious=true' : ''
+    return makeRequest('GET', `/api/device-screen${query}`)
```

Comment:
Curious - any reason for doing data fetching through SWR only for `useDeviceScreen` and not for `getDeviceScreen`? Could we update the names of these methods to make it more clear how they are different?

Reply:
I think `use*` and `get*` are the conventions for a hook vs async method right? That's essentially the difference here

Reply:
Yep, [hooks always starts with `use`](https://reactjs.org/docs/hooks-rules.html) but I guess the API is just a little bit confusing (to me) as to why both functions there. Is the `getDeviceScreen` there just so we can make the request outside of a hook/component where the `useDeviceScreen` hook can't be called?

Reply:
> Is the getDeviceScreen there just so we can make the request outside of a hook/component where the useDeviceScreen hook can't be called?

Yes that's exactly right. Let me know if there's a better way to name this?